prov/rxm/src/rxm_conn.c

```diff
@@ -87,6 +87,10 @@ static void rxm_close_conn(struct rxm_conn *conn)
 	}
 	fi_close(&conn->msg_ep->fid);
 	rxm_flush_msg_cq(conn->ep);
+
+	if ((!conn->ep->srx_ctx) && (conn->state != RXM_CM_CONNECTED))
+		ofi_bufpool_reset(conn->ep->rx_pool);
```

Comment:
Isn't the rx_pool used for all connections.  The kills the pool, even if the buffers are being used for connections that are still active.

Reply:
Agree. The preposted recv buffers should be freed when `msg_ep` is closed. That means a proper fix would be in the base provider instead of rxm. For example, in the verbs provider, the send queue is flushed when the endpoint is closed, but the receive queue is not.

Reply:
rxm_prepost_recv is allocating RX buffer from ep->rx_pool for msg_ep.  On a connection failure msg_ep is freed but leaving behind allocated RX buffer in ep->rx_pool .  ep->rx_pool grows from there on subsequent connection request.
Destroying ep->rx_pool was the first thing I tried.   To get that working, it would require a great deal of change and impacting other providers as well. I also looked into calling rxm_prepost_recv only once but still need to post recv to msg_ep so that didn't work out either that's why I ended up resetting the rx_pool.  Another approach is to track buffers allocated to msg_ep and only return those back to the pool but I would need to keep a list pointer somewhere, most likely in msg_ep which is a fid_ep or rxm_conn structure.  This is the least impactful patch I could come up with.  I don't like resetting the pool either but it was the easiest path.

Reply:
I'm looking at verbs provider to see what happens on vrb_ep_close

Reply:
A thought that occurred to me is that if the connection never formed, the QP is likely only in the RTR state.  I don't know if that is supposed to make a difference, or if it changes anything done in the rdmacm.

Reply:
it shouldn't.  you can post recv as soon as QP is created and if buffers are not flushed you cannot track resources correctly.  There is something missing.  I'm trying to verify verbs provider with CVL as I know it does flush.

Reply:
We need to check the call flow to see if the QP was ever transitioned into the error state.  If it was simply destroyed, it wouldn't flush.  If we called rdma_disconnect(), that will transition into the error state.  But if we were never connected at the OFI level, we're probably not calling that and going directly to rdma_destroy_ep(), which just closes the EP.